src/main/java/uk/gov/justice/hmpps/offenderevents/services/EventRetrievalService.java

```diff
@@ -79,4 +83,56 @@ public void pollEvents(final LocalDateTime currentTime) {
             log.warn("Skipping Event Retrieval as start after end, start = {}, end = {}", startTime, endTime);
         }
     }
+
+    /**
+     * For testing run a poll for the immediately ended minute, repeat 1 min later and compare:
+     * <ol>
+     *  <li>run for the latest minute (POLL_NAME_TEST.time -> now - 1 second) remembering time range and count, A
+     *  <li>run for [PREVIOUS_POLL_NAME_TEST -> POLL_NAME_TEST] time range remembering time range and count, B
+     *  <li>compare count B with POLL_NAME_TEST.count - should be the same
+     *  <li>store count A in POLL_NAME_TEST.count
+     *  <li>move POLL_NAME_TEST.time to PREVIOUS_POLL_NAME_TEST.time
+     *  <li>store A end time in POLL_NAME_TEST.time
+     * </ol>
+     */
+    public void runTestPolls(final LocalDateTime now) {
+        final LocalDateTime endTimeA = now.minus(1, SECONDS);
+
+        repository.findById(POLL_NAME_TEST).ifPresentOrElse(
+                test -> {
+                    final var previousTest = repository.findById(PREVIOUS_POLL_NAME_TEST).orElseThrow();
+                    final var eventsA = externalApiService.getEvents(test.getNextStartTime(), endTimeA);
+                    final var countA = eventsA.size();
+                    log.debug("runTestPolls(): A interval {} to {}, count {}", test.getNextStartTime(), endTimeA, countA);
+
+                    final var eventsB = externalApiService.getEvents(previousTest.getNextStartTime(), test.getNextStartTime());
+                    final var countB = eventsB.size();
+                    log.debug("runTestPolls(): B interval {} to {}, count {}", previousTest.getNextStartTime(), test.getNextStartTime(), countB);
+
+                    if (countB != test.getNumberOfRecords()) {
+                        log.warn("runTestPolls(): Found different counts, original={}, repeat={}, events {}", test.getNumberOfRecords(), countB, eventsB);
```

Comment:
not sure if outputting all the events here will help?  You won't be able to work out from them all which ones are new will you?

Reply:
i.e. what will be the next course of action if they don't match up?

Reply:
true but this time i was worried about too much log data. Could adjust depending on initial results